### DISCUSSION TITLE
AWM integration

### DISCUSSION THREAD

**Discussion by hyperantoine** [ava-labs/hypersdk#1244] [Chat]
I have a couple of questions on HyperSDK AWM support:
1) What will be the latency on the relayer setup? As HyperSDK is built for 100k TPS and low-latency, could the current 2 seconds that we observe be improved?
2) AWM integration inside actions was temporarily removed, but could you imagine in the future someone on the C-chain, bridging to an HyperVM, performing an action and then bridging back to the C-Chain in a single instruction?
3) Is there any plans to implement Teleporter at the VM level? I guess all HyperVMs could benefit from a standard when using bridged AWM tokens.

**Comment by aaronbuchwald** [comment]
AWM latency depends on:

1. Time from send message tx issuance to acceptance
2. Latency to aggregate a warp signature after acceptance
3. Time to deliver and accept the transaction delivering that transaction on the destination chain

The dominating factor for C-Chain Warp latency is the 2s block time, so this should be reduced on the HyperSDK side, but will still see an average of 1s latency when either sending/receiving a message on the C-Chain.

Yes, AWM integration will be added back to the HyperSDK when it's closer to production. The team's focus has shifted to improving the DevX and productionizing the code and interoperability support will be added back as a customizable network extension for any HyperVM to use out of the box.

There are no design plans to implement the exact Teleporter spec at the HyperVM or HyperSDK level. Teleporter is specific to the EVM, so integrating with any or all HyperVMs will depend on the exact VM implementation.

**Reply by hyperantoine** [reply]
Thanks!
